task.go

```diff
@@ -315,6 +315,7 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 		return nil, errors.Wrapf(errdefs.ErrFailedPrecondition, "task must be stopped before deletion: %s", status.Status)
 	}
 	if t.io != nil {
+		t.io.Close()
 		t.io.Cancel()
```

Comment:
Maybe cio_windows.go should implement the Cancel function to close the pipe listener as well.😀

Reply:
Not sure which ``cio_windows.go`` you refer to, but this seems to be ``containerd/pkg/cri/io/container_
io.go``.

Reply:
`pkg/cri/io/helpers_windows.go` doesn't use `ctx` for the pipe so that the `cancel` context doesn't work here. And the `copyIO` in `cio/cio_windows.go` doesn't use `ctx` either.